lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm_2019/ashrae_90_1_prm_2019.Model.rb

```diff
@@ -55,4 +55,264 @@ def model_get_percent_of_surface_range(model, wwr_parameter)
+      matching_objects << object
+    end
+
+    # If capacity was specified, narrow down the matching objects
+    unless capacity.nil?
+      # Skip objects that don't have fields for minimum_capacity and maximum_capacity
+      matching_objects = matching_objects.reject { |object| !object.key?('minimum_capacity') || !object.key?('maximum_capacity') }
+
+      # Skip objects that don't have values specified for minimum_capacity and maximum_capacity
+      matching_objects = matching_objects.reject { |object| object['minimum_capacity'].nil? || object['maximum_capacity'].nil? }
+
+      # Round up if capacity is an integer
+      if capacity == capacity.round
+        capacity += (capacity * 0.01)
+      end
+
+      # Skip objects whose the minimum capacity is below or maximum capacity above the specified capacity
+      matching_capacity_objects = matching_objects.reject { |object| capacity.to_f <= object['minimum_capacity'].to_f || capacity.to_f > object['maximum_capacity'].to_f }
+
+      # If no object was found, round the capacity down in case the number fell between the limits in the json file.
+      if matching_capacity_objects.size.zero?
+        capacity *= 0.99
+        # Skip objects whose minimum capacity is below or maximum capacity above the specified capacity
+        matching_objects = matching_objects.reject { |object| capacity.to_f <= object['minimum_capacity'].to_f || capacity.to_f > object['maximum_capacity'].to_f }
+      else
+        matching_objects = matching_capacity_objects
+      end
+    end
+    # If volume was specified, narrow down the matching objects
+    unless volume.nil?
+      # Skip objects that don't have fields for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| !object.key?('minimum_storage') || !object.key?('maximum_storage') }
+
+      # Skip objects that don't have values specified for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| object['minimum_storage'].nil? || object['maximum_storage'].nil? }
+      # Round up if volume is an integer
+      if volume == volume.round
+        volume += (volume * 0.01)
+      end
+      # Skip objects whose the minimum volume is below or maximum volume above the specified volume
+      matching_volume_objects = matching_objects.reject { |object| volume.to_f <= object['minimum_storage'].to_f || volume.to_f > object['maximum_storage'].to_f }
+
+      # If no object was found, round the volume down in case the number fell between the limits in the json file.
+      if matching_volume_objects.size.zero?
+        volume *= 0.99
+        # Skip objects whose minimum volume is below or maximum volume above the specified volume
+        matching_objects = matching_objects.reject { |object| volume.to_f <= object['minimum_storage'].to_f || volume.to_f > object['maximum_storage'].to_f }
+      else
+        matching_objects = matching_volume_objects
+      end
+    end
```

Comment:
I still don't quite understand why the volume needs to code the same as the capacity. It is based on the data form? You can coordinate with Jeremy on the new PRM data.
In addition, I cannot find a water heater data file in the repo that has volume. Could you check it again?

Reply:
Now we are using a new model_find_object function in Standards.Model.rb (line 2509) developed by Jeremy.